synapse/storage/databases/main/appservice.py

```diff
@@ -172,7 +174,12 @@ async def set_appservice_state(self, service, state) -> None:
             "application_services_state", {"as_id": service.id}, {"state": state}
         )
 
-    async def create_appservice_txn(self, service, events):
+    async def create_appservice_txn(
+        self,
+        service: ApplicationService,
+        events: List[EventBase],
```

Comment:
Thanks for adding types here! Can you remove them from the docstring? And we should probably add the return type while we're at it (and remove it from below).